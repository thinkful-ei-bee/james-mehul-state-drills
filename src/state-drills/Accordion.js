import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {
        sections: [] 
      }
   
    state = {
      currentSection: null,
    }

      renderButtons () {
         return this.props.sections.map((section, index) => (
          <li><button key={index} onClick={() => this.handleButtonClick(index)}>
            {section.title}
          </button>
          <p>{this.renderContent(index)}</p>
          </li>
        ))
      }

      handleButtonClick(index) {
          console.log("here");
        this.setState({ currentSection: index })
      }

      renderContent(index) {
       if(index === this.state.currentSection){
        const currentTab = this.props.sections[this.state.currentSection]
        return (
          <p className = "content">
            {currentTab.content}
          </p>
        )
       }
       else{
           return
       }
      }

    render() {
            return (
               <ul>
                 {!!this.props.sections.length && this.renderButtons()}
               </ul>
            )
  }
}

export default Accordion