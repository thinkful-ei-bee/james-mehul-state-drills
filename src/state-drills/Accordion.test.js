import { shallow } from 'enzyme'
import Accordion from './Accordion'
import React from 'react'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom'
import sections from '../sectionIndex';




describe(`Tabs Component`, () => {

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
      })
    

    it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the buttons and no tabs by default', () => {
    const wrapper = shallow(<Accordion sections = {sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens 1st  tab and closes any other tab', () => {
    const wrapper = shallow(<Accordion sections = {sections}/>)
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens 2nd  tab and closes any other tab', () => {
    const wrapper = shallow(<Accordion sections = {sections}/>)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens 3rd  tab and closes any other tab', () => {
    const wrapper = shallow(<Accordion sections = {sections}/>)
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })



  

})