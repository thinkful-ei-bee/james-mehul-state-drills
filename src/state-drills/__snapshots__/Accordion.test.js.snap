// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component closes the first tab and opens any clicked tab 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      <p
        className="content"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </p>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`Tabs Component opens 1st  tab and closes any other tab 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      <p
        className="content"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </p>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`Tabs Component opens 2nd  tab and closes any other tab 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      <p
        className="content"
      >
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </p>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`Tabs Component opens 3rd  tab and closes any other tab 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      <p
        className="content"
      >
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </p>
  </li>
</ul>
`;

exports[`Tabs Component renders empty given no tabs 1`] = `<ul />`;

exports[`Tabs Component renders the buttons and no tabs by default 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`renders empty given no tabs 1`] = `<ul />`;
